posydon/popsyn/norm_pop.py

```diff
@@ -0,0 +1,218 @@
+"""Compute the underlying stellar population weight of each model for a given simulation."""
+
+__authors__ = [
+    "Devina Misra <devina.misra@unige.ch>",
+    "Simone Bavera <Simone.Bavera@unige.ch>",
+    "Konstantinos Kovlakas <Konstantinos.Kovlakas@unige.ch>",
+    "Matthias Kruckow <Matthias.Kruckow@unige.ch>",
+    "Dimitris Souropanis <dsouropanis@ia.forth.gr>",
+    "Max Briel <max.briel@gmail.com>",
+]
+
+import numpy as np
+from posydon.popsyn import independent_sample
+from scipy.integrate import quad, nquad
+from posydon.utils.posydonwarning import Pwarn
+from posydon.popsyn.distributions import flat_mass_ratio, Sana12Period
+import posydon.popsyn.IMFs as IMFs
+
+def get_IMF_pdf(kwargs):
+    '''get the IMF pdf function'''
+    
+    primary_mass_scheme = kwargs.get('primary_mass_scheme', '')
+    scheme_kwargs = kwargs.get(primary_mass_scheme, {})
+    try:
+        # dynamically retrieve the IMF class from the IMFs module
+        imf_class = getattr(IMFs, primary_mass_scheme)
+        imf = imf_class(m_min=kwargs['primary_mass_min'],
+                        m_max=kwargs['primary_mass_max'],
+                        **scheme_kwargs)
+        IMF_pdf = imf.pdf
+    except AttributeError:
+        # if not found, default to a flat distribution
+        IMF_pdf = lambda m1: 1
+        
+    return IMF_pdf
+
+def get_mass_ratio_pdf(kwargs):
+    """Function that returns the mass ratio PDF function"""
+    if kwargs['secondary_mass_scheme'] == 'flat_mass_ratio' and ('q_min' not in kwargs and 'q_max' not in kwargs):
+        # flat mass ratio, where bounds are dependent on m1 and min/max m2
+        # and q_min = 0.05, q_max = 1
+        def get_pdf_for_m1(m1):
+            m1 = np.atleast_1d(m1)
+            minimum = np.max([kwargs['secondary_mass_min'] / m1, np.ones(len(m1))*0.05], axis=0)
+            maximum = np.min([kwargs['secondary_mass_max'] / m1, np.ones(len(m1))], axis=0)
+            q_dist = lambda q: np.where((q >= minimum) & (q <= maximum), 1/(maximum - minimum), 0)
+            return q_dist
+        q_pdf = lambda q, m1: get_pdf_for_m1(m1)(q)
+    else:
+        q_pdf = lambda q, m1=None: np.where((q > 0.0) & (q<=1.0), 1, 0)
```

Comment:
```suggestion
    elif kwargs['secondary_mass_scheme'] == 'flat_mass_ratio':
        q_pdf = lambda q, m1=None: np.where((q > kwargs['q_min'])
                                            & (q <= kwargs['q_max']),
                                            1/(kwargs['q_max']
                                               - kwargs['q_min']), 0)
    else:
        Pwarn("The secondary_mass_scheme is not defined use a flat mass ratio "
              "in [0,1].", "UnsupportedModelWarning")
        q_pdf = lambda q, m1=None: np.where((q > 0.0) & (q<=1.0), 1, 0)
```
P.S.: please don't forget to import `Pwarn`.